dimod/reference/samplers/identity_sampler.py

```diff
@@ -97,6 +99,12 @@ def sample(self, bqm, *args, **kwargs):
 
 
         """
+        for kw in kwargs.copy():
+            if kw not in self.parameters:
+                msg = "Ignoring unknown kwarg: {!r}".format(kw)
+                warnings.warn(msg, RuntimeWarning)
```

Comment:
Instead of repeating the same pattern, perhaps extract (some variant or portion of) it to `dimod.Sampler` base class?